src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Syntax/IStartTagSyntaxNode.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.AspNetCore.Razor.Language.Syntax;
+
+internal interface IStartTagSyntaxNode
```

Comment:
@dotnet/razor-compiler if you don't like this, then you have to either fix the syntax node heirachy in Razor, or implement extensions and roles in C#. I don't make the rules.